src/exec/mod.rs

```diff
@@ -46,10 +47,14 @@ use self::{
 ///
 /// Returns the [`ExitReason`] of the command and a function that restores the default handler for
 /// signals once its called.
-pub fn run_command(
-    options: &impl RunOptions,
-    env: Environment,
-) -> io::Result<(ExitReason, impl FnOnce())> {
+pub fn run_command(options: &impl RunOptions, env: Environment) -> io::Result<ExecOutput> {
+    match run_command_internal(options, env)? {
+        ProcessOutput::SudoExit { output } => Ok(output),
+        ProcessOutput::ChildExit => _exit(1),
```

Comment:
> The child processes shouldn't try to flush the parent's IO streams.

could you add this as a comment? sounds like only the original "parent' process will call `exit` and the other 2 processes will call `_exit`. which IO streams are we trying to not "over flush"? stdin, stdout, stderr? or some other file / pipe?

does the rationale also hold when some processes get their own PTY? what about when there are only 2 processes (`!no_pty` exec mode)?

Reply:
just `stdin`, `stdout` and `stderr` yes.

The issue when the process gets a pseudoterminal is that if sudo is part of a pipeline it will inherit the piped IO streams nevertheless so it could still happen. If sudo is not part of a pipeline, the command will just use the pseudoterminal and it would be fine if the command flushes it AFAIK.

When `!use_pty` is enabled, the child process just inherits the IO streams of the parent and no extra set up is done so they could be flushed on `exit` anyway.